src/lib/tab.ts

```diff
@@ -329,11 +346,14 @@ export const open = async (entry: Entry, type: TabType = detectType(entry)): Pro
     if (!tab) {
         // tab doesn't exist, create
         try {
+            const existing = Array.from(get(tabs).values()).filter((t) => t.position === TabPosition.PRIMARY_CENTER);
```

Comment:
this should be the same logic as in the `openUnscoped` function I think?

also I would move this logic up to `update` that will replace a "default" value (such as -1) with a proper index, so you don't have to duplicate the logic (you also have access to `$tabs` directly there)